system/src/system_network_diagnostics.cpp

```diff
@@ -66,23 +67,58 @@ class NetworkCache
 public:
     static const system_tick_t NETWORK_INFO_CACHE_INTERVAL = 1000;
 
-    const Signal* getSignal()
+    const Signal* getSignal(bool getPrimarySignal = true)
     {
-#if Wiring_WiFi || Wiring_Cellular
+        bool refreshCachedValues = false;
         system_tick_t m = millis();
-        if (ts_ == 0 || (m - ts_) >= NETWORK_INFO_CACHE_INTERVAL)
-        {
-#if Wiring_Cellular
-            sig_ = Cellular.RSSI();
-#elif Wiring_WiFi
-            sig_ = WiFi.RSSI();
-#endif
+        if (ts_ == 0 || (m - ts_) >= NETWORK_INFO_CACHE_INTERVAL) {
             ts_ = millis();
+            refreshCachedValues = true;
+        }
+
+// TODO: Better differentiator for "ACM"
+#if PLATFORM_ID == PLATFORM_MSOM
```

Comment:
It feels like I should create a platform `#define` for ACM support. ideas for a name?

Reply:
There is `HAL_PLATFORM_NCP_COUNT` but that will include Tracker platforms. Perhaps a brand new define like `HAL_PLATFORM_AUTOMATIC_CONNECTION_MANAGEMENT`?